matrix_client/crypto/olm_device.py

```diff
@@ -19,13 +19,36 @@ class OlmDevice(object):
         device_id (str): Must match the one used when logging in.
     """
 
+    _olm_algorithm = 'm.olm.v1.curve25519-aes-sha2'
+    _megolm_algorithm = 'm.megolm.v1.aes-sha2'
+    _algorithms = [_olm_algorithm, _megolm_algorithm]
+
     def __init__(self, api, user_id, device_id):
         self.api = api
         check_user_id(user_id)
         self.user_id = user_id
         self.device_id = device_id
         self.olm_account = olm.Account()
         logger.info('Initialised Olm Device.')
+        self.identity_keys = self.olm_account.identity_keys
+        self.one_time_key_counts = {}
+
+    def upload_identity_keys(self):
+        """Uploads this device's identity keys to HS.
+
+        This device must be the one used when logging in.
+        """
+        device_keys = {
+            'user_id': self.user_id,
+            'device_id': self.device_id,
+            'algorithms': self._algorithms,
+            'keys': {'{}:{}'.format(alg, self.device_id): key
+                     for alg, key in self.identity_keys.items()}
+        }
+        self.sign_json(device_keys)
+        ret = self.api.upload_keys(device_keys=device_keys)
+        self.one_time_key_counts = ret['one_time_key_counts']
```

Comment:
This happens when the server has no more key of that type. So when a key is missing in `self.one_time_key_counts`, we infer that it means a count of 0 (hence we do `self.one_time_key_counts.get(type, 0)`). I'm not really sure keeping this dict is good, even though it is the solution with the less overhead.  
Would you prefer if there was two interger attribute for the counts, and two more for the target_counts? This would allow to get rid of the weird dicts, but if there are new types of one-time keys later it could lead to a lot of attributes...